main.js

```diff
@@ -49,5 +49,5 @@ exports._unrefActive = exports.active = function(item) {
 
 // setimmediate attaches itself to the global object
 require("setimmediate");
-exports.setImmediate = setImmediate;
-exports.clearImmediate = clearImmediate;
+exports.setImmediate = self && self.setImmediate || global && global.setImmediate || window && window.setImmediate || setInterval;
```

Comment:
Please add a comment in code about this being the order that `setimmediate` tries to attach to the global. Something about this being occasionally necessary in exotic environments might be nice as a comment as well.

Also, it looks like the library tries: `self`, `global`, and `this`. To be more accurate, shouldn't `window` be replaced with `this` instead?

Reply:
I have addressed the comments and updated the PR, thinking about it though, `require` is a function (and webpack wraps `require`s in its own functions) wouldn't that mean that `this` would be the require function itself?

Reply:
Hmm... I would hope Webpack is okay with this, as I feel like many other things would break if it was not.

I tried a super-quick test with Webpack, and it appears to work with you latest patch, so let's proceed.